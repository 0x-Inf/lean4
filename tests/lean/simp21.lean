variable vec    : Nat â†’ Type
rewrite_set simple
add_rewrite Nat::add_assoc Nat::add_zeror eq_id : simple
variable n : Nat
(*
local t = parse_lean([[ vec n = vec (n + 0) ]])
print(t)
print("===>")
local t2, pr = simplify(t, "simple")
print(t2)
print(pr)
*)