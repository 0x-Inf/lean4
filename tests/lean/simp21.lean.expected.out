  Set: pp::colors
  Set: pp::unicode
  Assumed: vec
  Assumed: n
vec n = vec (n + 0)
===>
‚ä§
trans (congr2 (eq (vec n)) (congr2 vec (Nat::add_zeror n))) (eq_id (vec n))
